components/raftstore/src/store/fsm/apply.rs

```diff
@@ -960,19 +976,20 @@ where
 
             if apply_ctx.yield_high_latency_operation && has_high_latency_operation(&cmd) {
                 self.priority = Priority::Low;
-                if apply_ctx.priority != Priority::Low {
-                    apply_ctx.commit(self);
-                    return ApplyResult::Yield;
-                }
             }
-            if should_write_to_engine(&cmd) || apply_ctx.kv_wb().should_write_to_engine() {
+            let need_sync = should_write_to_engine(&cmd)
+                && self.last_sync_applied_index != self.apply_state.get_applied_index();
+            if need_sync || apply_ctx.kv_wb().should_write_to_engine() {
                 apply_ctx.commit(self);
                 if let Some(start) = self.handle_start.as_ref() {
                     if start.elapsed() >= apply_ctx.yield_duration {
                         return ApplyResult::Yield;
                     }
                 }
             }
+            if self.priority != apply_ctx.priority {
+                return ApplyResult::Yield;
+            }
```

Comment:
Before `Yield`, the `commit` must be called otherwise the data may be wrong due to reordering.

Reply:
OK, I will add it.